src/compute/src/render/sinks.rs

```diff
@@ -68,18 +69,18 @@ where
             let (permutation, thinning) = permutation_for_arrangement(&key, unthinned_arity);
             let mut mfp = MapFilterProject::new(unthinned_arity);
             mfp.permute(permutation, thinning.len() + key.len());
-            let (collection, _err_collection) =
-                bundle.as_collection_core(mfp, Some((key.clone(), None)));
-            collection
+            bundle.as_collection_core(mfp, Some((key.clone(), None)))
         };
 
-        let collection = apply_sink_envelope(sink, &sink_render, collection);
-
-        // TODO(benesch): errors should stream out through the sink,
-        // if we figure out a protocol for that.
+        let ok_collection = apply_sink_envelope(sink, &sink_render, ok_collection);
```

Comment:
Should we have a TODO here to also apply an envelope to `err_collection`?

Reply:
The Kafka sink's days in `compute` are numbered, as it will move to storage as part of the external sinks feature. `apply_sink_envelope` (and a bunch of other code) only exists to support the Kafka sink, so we can remove it at that point too. So if we want to add a TODO, I'd rather have one for removing the envelope from the `ok_collection`.

Reply:
Thanks, that makes sense. Feel free to add a TODO, but if its days are numbered, then it's fine with me not to!